### PR TITLE
documentation

### PR DESCRIPTION
**Pull request by Rahgooy**

### CODE REVIEW

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# Spatial Role Labeling
+In paper [1], the task of spatial role labeling is introduced and an annotation scheme proposed that is language-independent and facilitates the application of machine learning techniques. The framework consists of a set of spatial roles. For a basic example, in the following sentence:
+
+
+Give me [the gray book] (trajector) [on] (spatialindicator) [the big table] (landmark).
+
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md:6]: here markdown shows the items in brackets as links with null pointer!

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+The phrase headed by the token book is referring to a trajector object, the phrase headed by the token table is referring to the role of a landmark and these are related by the spatial expression on denoted as spatial indicator. The spatial indicator (often a preposition) establishes the type of spatial relation.
+
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md]: you can add `The model has a classification setting that classifies candidate spatial relations which are a set of candidate triplets to positive and negative.  However, the model is interesting in that is does a lot of effort for feature engineering which is much easier to do using Saul here. (some thing kine that?)

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+Analogous to semantic role labeling, spatial role labeling is defined as the task of automatic labeling of words or phrases in a sentence with a set of spatial roles. More speciﬁcally, it involves identifying and classifying the spatial arguments of the spatial expressions mentioned in a sentence.
+
+
+## SpRL using Saul
+In this example, we implement the best performing algorithm for [task 3 of SemEval 2012](https://www.cs.york.ac.uk/semeval-2012/task3.html) proposed in [2] using Saul.
+
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md:16]: you can add `The model has a classification setting that classifies candidate spatial relations which are a set of candidate triplets to positive and negative. However, the model is interesting in that is does a lot of effort for feature engineering which is much easier to do using Saul here. (some thing kine that?)

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      })
+    })
+```
+`SpRLDataReader` is the xml data reader which loads data from the dataset. `getDocumentSentences` splits the sentences and returns a list of tuples. Each tuple contains a sentence and it's character offset in the document. 
+`TextAnnotationFactory.createTextAnnotation` uses the NLP-Pipeline to generate feature rich `TextAnnotation` for each sentence. `getRelations` returns the relations that correspond to the current sentence. 
+
+
+### Defining the `DataModel`
+In order to identify spatial relations, the authors of [2] used a simple method.
+A set of triples used to represent the spatial relations. They generated many candidates and trained a classifier to classify them. So here is the [`DataModel`](RobertsDataModel.scala) :
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md:41]: I see you have some of the info i wrote up in here, so keep them once, if you prefer just here is good then.

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+## Running
+To run the main app with default properties:
+
+```
+sbt "project saulExamples" "runMain edu.illinois.cs.cogcomp.saulexamples.nlp.SpatialRoleLabeling.SpRLApp"
+```
+
+## References
+[1] Kordjamshidi, Parisa, Steven Bethard, and Marie-Francine Moens. "SemEval-2012 task 3: Spatial role labeling." Proceedings of the First Joint Conference on Lexical and Computational Semantics. Association for Computational Linguistics, 2012.
+
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/README.md:111]: Add the ACM-TSLP paper also?

**Comment by kordjamshidi** [comment]
why all these files have been changed in this PR? shouldn't this be about documentation only?

**Comment by Rahgooy** [comment]
good question!

**Pull request merged**